flexmeasures/api/v3_0/sensors.py

```diff
@@ -154,3 +189,347 @@ def get_data(self, response: dict):
+            index=pd.date_range(
+                start_of_schedule, end_of_schedule, freq=resolution, closed="right"
+            ),  # note that target values are indexed by their due date (i.e. closed="right")
+        )
+        for target in kwargs.get("targets", []):
+
+            # get target value
+            if "value" not in target:
+                return ptus_incomplete("Target missing value parameter.")
+            try:
+                target_value = float(target["value"])
+            except ValueError:
+                extra_info = "Request includes empty or ill-formatted target value(s)."
+                current_app.logger.warning(extra_info)
+                return ptus_incomplete(extra_info)
+            if unit == "kWh":
+                target_value = target_value / 1000.0
+
+            # get target datetime
+            if "datetime" not in target:
+                return invalid_datetime("Target missing datetime parameter.")
+            else:
+                target_datetime = target["datetime"]
+                if target_datetime is None:
+                    return invalid_datetime(
+                        "Cannot parse target datetime string %s as iso date"
+                        % target["datetime"]
+                    )
+                if target_datetime.tzinfo is None:
+                    current_app.logger.warning(
+                        "Cannot parse timezone of target 'datetime' value %s"
+                        % target["datetime"]
+                    )
+                    return invalid_timezone(
+                        "Target datetime should explicitly state a timezone."
+                    )
+                if target_datetime > end_of_schedule:
+                    return invalid_datetime(
+                        f'Target datetime exceeds {end_of_schedule}. Maximum scheduling horizon is {current_app.config.get("FLEXMEASURES_PLANNING_HORIZON")}.'
+                    )
+                target_datetime = target_datetime.astimezone(
+                    soc_targets.index.tzinfo
+                )  # otherwise DST would be problematic
+
+            # set target
+            soc_targets.loc[target_datetime] = target_value
+
+        job = create_scheduling_job(
+            sensor.id,
+            start_of_schedule,
+            end_of_schedule,
+            resolution=resolution,
+            belief_time=prior,  # server time if no prior time was sent
+            soc_at_start=value,
+            soc_targets=soc_targets,
+            soc_min=soc_min,
+            soc_max=soc_max,
+            roundtrip_efficiency=roundtrip_efficiency,
+            enqueue=True,
+        )
+
+        # Store new soc info as GenericAsset attributes
+        # sensor.generic_asset.set_attribute("soc_datetime", start_of_schedule.isoformat())
+        # sensor.generic_asset.set_attribute("soc_udi_event_id", event_id)
+        # sensor.generic_asset.set_attribute("soc_in_mwh", value)
+
+        db.session.commit()
```

Comment:
This `db` is the postgres database, not Redis. I might have overlooked something, but before we uncomment the lines above, we don't need to commit this session right?